src/world/World.php

```diff
@@ -677,7 +677,7 @@ public function addSound(Vector3 $pos, Sound $sound, ?array $players = null) : v
 
 		$pk = $ev->getSound()->encode($pos);
 		if(count($pk) > 0){
-			if($players === $this->getViewersForPosition($pos)){
+			if($players === $ev->getRecipients()){
```

Comment:
This is not the correct fix. The correct solution would be to overwrite $players with the result of getRecipients() before entering this code.